src/cross_toml.rs

```diff
@@ -234,4 +311,218 @@ mod tests {
+            Target::Custom {
+                triple: "target2".to_string(),
+            },
+            CrossTargetConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR2_PRECEDENCE".to_string()],
+                    volumes: vec!["VOL2_ARG_PRECEDENCE".to_string()],
+                },
+                xargo: Some(false),
+                build_std: Some(false),
+                image: Some("test-image2-precedence".to_string()),
+                runner: None,
+            },
+        );
+        targets2.insert(
+            Target::Custom {
+                triple: "target3".to_string(),
+            },
+            CrossTargetConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR3".to_string()],
+                    volumes: vec!["VOL3_ARG".to_string()],
+                },
+                xargo: Some(false),
+                build_std: Some(true),
+                image: Some("test-image3".to_string()),
+                runner: None,
+            },
+        );
+
+        // Defines the base config
+        let cfg1 = CrossToml {
+            targets: targets1,
+            build: CrossBuildConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR1".to_string(), "VAR2".to_string()],
+                    volumes: vec![],
+                },
+                build_std: Some(true),
+                xargo: Some(true),
+                default_target: None,
+            },
+        };
+
+        // Defines the config that is to be merged into cfg1
+        let cfg2 = CrossToml {
+            targets: targets2,
+            build: CrossBuildConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR3".to_string(), "VAR4".to_string()],
+                    volumes: vec![],
+                },
+                build_std: None,
+                xargo: Some(false),
+                default_target: Some("aarch64-unknown-linux-gnu".to_string()),
+            },
+        };
+
+        // Defines the expected targets after the merge
+        let mut targets_expected = HashMap::new();
+        targets_expected.insert(
+            Target::BuiltIn {
+                triple: "aarch64-unknown-linux-gnu".to_string(),
+            },
+            CrossTargetConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR1".to_string()],
+                    volumes: vec!["VOL1_ARG".to_string()],
+                },
+                xargo: Some(false),
+                build_std: Some(true),
+                image: Some("test-image1".to_string()),
+                runner: None,
+            },
+        );
+        targets_expected.insert(
+            Target::Custom {
+                triple: "target2".to_string(),
+            },
+            CrossTargetConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec!["VAR2_PRECEDENCE".to_string()],
```

Comment:
I'm not sure this is how I would expect them to be merged.

`Cargo.toml` has the following:

```toml
[package.metadata.cross.target.target2.env]
passthrough = ["VAR2"]
volumes = ["VOL2_ARG"]
```

And `Cross.toml` has:

```toml
[target.target2.env]
passthrough = ["VAR2_PRECEDENCE"]
volumes = ["VOL2_ARG_PRECEDENCE"]
```

I would expect the result to be `["VAR2", "VAR2_PRECEDENCE"]` and `["VOL2", "VOL2_PRECEDENCE"]`, that is, non-scalar values get combined, and scalar values get overwritten.

`build-std` here is correct (it correctly overrides the `true` value with the `false` one, but I think the arrays should be merged, not over written.